To: vim-dev@vim.org
Subject: Patch 7.2.257
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.2.257
Problem:    With GTK 2.17 lots of assertion error messages.
Solution:   Remove check for static gravity. (Sebastian Droege)
Files:	    src/gui_gtk_f.c


*** ../vim-7.2.256/src/gui_gtk_f.c	2009-05-17 23:25:16.000000000 +0200
--- gui_gtk_f.c	2009-09-11 15:15:41.000000000 +0200
***************
*** 860,870 ****
  gtk_form_set_static_gravity(GdkWindow *window, gboolean use_static)
  {
  #ifdef HAVE_GTK2
!     gboolean static_gravity_supported;
! 
!     static_gravity_supported = gdk_window_set_static_gravities(window,
! 							       use_static);
!     g_return_if_fail(static_gravity_supported);
  #else
      XSetWindowAttributes xattributes;
  
--- 860,868 ----
  gtk_form_set_static_gravity(GdkWindow *window, gboolean use_static)
  {
  #ifdef HAVE_GTK2
!     /* We don't check if static gravity is actually supported, because it
!      * results in an annoying assertion error message. */
!     gdk_window_set_static_gravities(window, use_static);
  #else
      XSetWindowAttributes xattributes;
  
-- 
hundred-and-one symptoms of being an internet addict:
231. You sprinkle Carpet Fresh on the rugs and put your vacuum cleaner
     in the front doorway permanently so it always looks like you are
     actually attempting to do something about that mess that has amassed
     since you discovered the Internet.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\        download, build and distribute -- http://www.A-A-P.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
